peercrawler.py

```diff
@@ -82,6 +82,31 @@ def count_good_peers(self):
     def count_peers(self):
         return len(self.get_peers_copy())
 
+    def listen_incoming(self):
+        with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as s:
+            s.bind(("::", 7075))
+            s.listen()
+
+            while True:
+                connection, address = s.accept()
+                threading.Thread(target=self.handle_incoming, args=(connection, address), daemon=True).start()
+
+    def handle_incoming(self, connection, address):
+        print(f"Receiving connection from {address}")
+
+        header_raw = connection.recv(8)
+        header = message_header.parse_header(header_raw)
+
+        if header.msg_type != message_type(message_type_enum.keepalive):
```

Comment:
This is not enough, I would expect the first message to arrive to be either a handshake packet or a bootstrap request.

Reply:
Sorry, I looked at the first commit rather that the whole changlist.
I can see that the handshake is done on later commits. My mistake.